server/src/main/java/org/elasticsearch/action/admin/indices/shrink/ResizeType.java

```diff
@@ -12,5 +12,5 @@
  * The type of the resize operation
  */
 public enum  ResizeType {
-    SHRINK, SPLIT, CLONE;
+    SHRINK, SPLIT, CLONE, PREPARE_CLONE;
```

Comment:
As above, I do not believe this qualifies as a resize operation.

Reply:
I'm also not convinced the `indices.shrink` package is the right place for these classes.

Reply:
yeah sure, it make sense. Should I move it under `create` package?

Reply:
I think that would make sense as long as it doesn't introduce any problems with code sharing with the shrink/split/clone code.

Reply:
Since enum type of `CLONE` already sharing `CreateIndexClusterStateUpdateRequest` class, accepting `ResizeType` as param. Should I just use `CLONE` instead of `PREPARE_CLONE`? My two cent will be to create a new enum type to handle and distinguish this specific scenario instead of using `CLONE` in the `ResizeType` enum. Please let me know your thought on this?

Reply:
I'm not sure that I understand your question above. The `CreateIndexClusterStateUpdateRequest` accepts an optional `ResizeType` for resize operations, but for this new prepare_reindex_target operation, I believe that should be left `null` since it is not a resize operation. Does that answer your question?

Reply:
Yeah, agree. I was under the assumption, there is a 'cause' field I need to setup using this ENUM. I will use separate string like `prepare_clone_index` for that.

Reply:
> Yeah, agree. I was under the assumption, there is a 'cause' field I need to setup using this ENUM. I will use separate string like `prepare_clone_index` for that.

That sounds good although perhaps `prepare_reindex_target` rather than `prepare_clone_index`.